### PR TITLE
python311Packages.pyeclib: unstable-2022-03-11 -> 1.6.1

### PR DESCRIPTION
**Pull request by natsukium**
## Description of changes

Diff: https://github.com/openstack/pyeclib/compare/b50040969a03f7566ffcb468336e875d21486113...1.6.1

Changelog: https://github.com/openstack/pyeclib/blob/1.6.1/ChangeLog

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 261572` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>python310Packages.pyeclib</li>
    <li>python310Packages.pyeclib.dist</li>
    <li>python310Packages.swift</li>
    <li>python310Packages.swift.dist</li>
    <li>python311Packages.pyeclib</li>
    <li>python311Packages.pyeclib.dist</li>
    <li>python311Packages.swift</li>
    <li>python311Packages.swift.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/pyeclib/default.nix [hunk 1]
```diff
@@ -1,14 +1,23 @@
...
+, setuptools
+, six
+, pytestCheckHook
+}:
 
 buildPythonPackage rec {
   pname = "pyeclib";
-  version = "unstable-2022-03-11";
+  version = "1.6.1";
+  pyproject = true;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pyeclib/default.nix:14]: There is no pyproject file https://github.com/openstack/pyeclib/tree/1.6.1

Comment (NickCao) [reply pkgs/development/python-modules/pyeclib/default.nix:14]: `pyproject = true` can (and might be preferred) be used with legacy projects, see https://github.com/NixOS/nixpkgs/issues/253154

Comment (SuperSandro2000) [reply pkgs/development/python-modules/pyeclib/default.nix:14]: I don't think we need to jump ahead here until this becomes the default and then we don't need to set it.

**Comment by SuperSandro2000** [comment]
done as part of #338883

**Pull request closed without merging**